app/components/sessions/SessionsController.js

```diff
@@ -18,57 +18,72 @@ sessionsModule.config(['$stateProvider', function($stateProvider) {
 sessionsModule.controller('SessionsController',
     ['$mdDialog', '$sessionStorage', '$rootScope', 'ApiJsonFactory',
         function($mdDialog, $sessionStorage, $rootScope, ApiJsonFactory) {
-        var sc = this;
-        if ($sessionStorage.sessions === null ||
-            typeof($sessionStorage.sessions) == 'undefined')
-        {
-            $sessionStorage.sessions = [];
-        }
-        sc.Sessions = $sessionStorage.sessions;
-
-        if (sc.Sessions.length === 0) {
-            ApiJsonFactory.getJson('sessions')
-                .then(function (response) {
-                    sc.Sessions = response.data.sessions;
-                    $sessionStorage.sessions = sc.Sessions;
-                }, function (error) {
-                    console.error(error);
-                });
-        }
-        sc.duration = function(session) {
-            var start = DateUtils.getHourMin(session.begin);
-            var end = DateUtils.getHourMin(session.end);
+            var sc = this;
+            if ($sessionStorage.sessions === null ||
+                typeof($sessionStorage.sessions) == 'undefined')
+            {
+                $sessionStorage.sessions = [];
+            }
 
-            return start + ' - ' + end;
-        };
+            sc.Sessions = new Array(totalDays);
```

Comment:
Yeah cannot depend upon `totalDays` to be defined, u should inject it as a dependent module like you inject `$mdSidenav, $mdMedia, $sessionStorage, ApiJsonFactory`